### ISSUE TITLE
Feature Request: Out-Of-Bag Error should not require storing the model.

### ISSUE THREAD

**Issue by sebffischer** [mlr-org/mlr3#1158]
It is currently necessary to store the `$model` of a learner in order to access its `oob_error`.
An example for this is the random forest:

```{r}
library(mlr3verse)
rr = resample(tsk("iris"), rsmp("classif.ranger"), rsmp("holdout"), store_models = TRUE)
rr$aggregate(msr("oob_error"))
```

The code below will err if we don't specify `store_models = TRUE` because the out-of-bag error active binding extracts this information from `learner$model`, which will fail of the model is not stored.

Instead of making the out-of-bag error active binding access the learner's `$model`, we can instead support the private function `$.extract_oob_error()` which will add the out-of-bag error score to the learner's `$state`, which is even accessible when `store_models = FALSE`.
This is already implemented for internal validation scores and internal tuned values:

https://github.com/mlr-org/mlr3/blob/5ffcfeeb1879e756b0cbd26a60a75faaa46aab78/R/worker.R#L94-L100

**Comment by be-marc** [comment]
We could solve this with callbacks.

```r
library(mlr3learners)

task = tsk("pima")
learner = lrn("classif.ranger")
resampling = rsmp("cv", folds = 3)

callback = clbk("mlr3.score_measures", measures = msr("oob_error"))

rr = resample(task, learner, resampling = resampling, store_models = FALSE, callbacks = callback)
rr$data_extra

# [[1]]
# [[1]]$score_measures
# oob_error 
# 0.2597656 


# [[2]]
# [[2]]$score_measures
# oob_error 
# 0.2402344 


# [[3]]
# [[3]]$score_measures
# oob_error 
# 0.2480469 
```

**Comment by sebffischer** [comment]
this does not allow for tuning i assume?

**Comment by be-marc** [comment]
Yes, it would be cool if `$score()` and `$aggregate()` could recognize the return of the callback in `$data_extra` as performance scores. You could solve this with tuning callbacks but I would prefer a more elegant solution.

**Comment by be-marc** [comment]
@berndbischl decided that the output of a callback is an end point. So `$score()` and `$aggregate()` cannot work with it. In the tuning case, I would use a `CallbackTuning` that saves the models of a tuning iteration temporary, call `msr("oob_error")` on them and then throw the models away. Or you combine a `CallbackResample` with a `CallbackTuning`. The `CallbackResample`  extracts the oob error on the worker without storing the models. The `CallbackTuning` passes the oob error saved in `data_extra` to the archive. Close?

**Comment by sebffischer** [comment]
but we use the oob error tuning example in the book so we need to fix this

**Comment by sebffischer** [comment]
yes

**Comment by berndbischl** [comment]
lets put this in learner$state. @be-marc search for 10 min whether there are other things like this....
at least in "measures"
